app/transport/http/user.go

```diff
@@ -9,13 +10,27 @@ import (
 // CreateUser will handle user creation
 func (uh UserHandler) CreateUser() http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
-		usr := entities.User{}
+		user := entities.User{}
 
-		if err := usr.Validate(); err != nil {
+		if err := json.NewDecoder(req.Body).Decode(&user); err != nil {
+			uh.logger.Errorf("%+v\n", err)
 			return
 		}
 
-		if err := uh.usecase.Signup(usr); err != nil {
+		defer func() {
+			if err := req.Body.Close(); err != nil {
+				uh.logger.Warnf("%+v\n", err)
+			}
+
+		}()
+
+		if err := user.Validate(); err != nil {
+			uh.logger.Infof("%+v\n", err)
```

Comment:
Why `Infof`? It looks like a bad request, so we definitely should log an error via `Errorf`.

Reply:
Thanks, I was not certain about this case.